packages/core/src/hooks/useTransactionOverride.ts

```diff
@@ -0,0 +1,72 @@
+import { usePromiseTransaction } from './usePromiseTransaction'
+import { StoredTransaction } from '../providers/transactions/model'
+import { useEthers } from './useEthers'
+import { TransactionRequest } from '@ethersproject/providers'
+import { BigNumber, utils } from 'ethers'
+
+export interface overrideRespond {
+  request: TransactionRequest
+  error?: string
+}
+
+export function useTransactionOverride() {
+  const { chainId, library } = useEthers()
+  const { promiseTransaction, state } = usePromiseTransaction(chainId)
+
+  const overrideTransaction = async (
+    transaction: StoredTransaction,
+    transactionRequest: TransactionRequest
+  ): Promise<overrideRespond> => {
+    const request: TransactionRequest = { ...transactionRequest, nonce: transaction.transaction.nonce }
+    if (library?.connection.url === 'metamask') {
+      return { request, error: "Metamask doesn't support custom nonce" }
+    }
```

Comment:
Metamask doesn't support custom nonce injection so to avoid doubling transactions function returns without sending